api/src/main/scala/za/co/absa/loginsvc/rest/SecurityConfig.scala

```diff
@@ -49,9 +57,21 @@ class SecurityConfig {
         .and()
       .httpBasic()
 
+    //TODO: Neaten up checking for Config
+    if(KerberosConfig != null)
+      {
+        if(KerberosConfig.enableKerberos.isDefined)
+          {
+            val kerberos = new KerberosSPNEGOAuthenticationProvider(KerberosConfig)
+            val provider = new RestApiKerberosAuthenticationProvider(KerberosConfig.url, KerberosConfig.searchFilter, KerberosConfig.domain);
+            http.addFilterBefore(
+              RestApiKerberosAuthentication.spnegoAuthenticationProcessingFilter(
+                new ProviderManager(provider, kerberos.kerberosServiceAuthenticationProvider())),
+              classOf[BasicAuthenticationFilter])
+          }
+      }
+
     http.build()
```

Comment:
I am surprised that it is necessary to force the the `Kerberos...AuthProvider` this way. Isn't this just another `AuthenticationProvider` just like `ConfigUsersAuthenticationProvider` and `ActiveDirectoryLDAPAuthenticationProvider`?

Perhaps it could be provided the same way as those are in `AuthManagerConfig.authManager:47`?

Reply:
So I'm forcing the Authmanager here so that I can try and debug any issues with the kerberos auth specifically. Just trying to make it easier to find the above issue that is being run into.